deserialize.go

```diff
@@ -16,9 +17,10 @@ func Deserialize(data []byte) (list List, err error) {
 	err = binary.Read(buffer, binary.BigEndian, &value)
 
 	if err != nil {
+		err = fmt.Errorf("could not read the input %v", value)
```

Comment:
You want to return your own error, as it will capture all the possible upstream binary.Read(...) errors in a giant net and allows us to test that an error of this text was returned.